storagenode/piecestore/endpoint.go

```diff
@@ -123,6 +129,8 @@ func (endpoint *Endpoint) Delete(ctx context.Context, delete *pb.PieceDeleteRequ
 	atomic.AddInt32(&endpoint.liveRequests, 1)
 	defer atomic.AddInt32(&endpoint.liveRequests, -1)
 
+	endpoint.pingStats.WasPinged(time.Now())
+
```

Comment:
After an uplink successfully uploads/deletes/etc a piece for a storagenode do you know if the satellite updates the `nodes.last_contact_success` table for that storagenode?

Reply:
Nope. These messages are most likely between uplink and storage node. The satellite creates an order limit but will not try to talk to the storage node and not update `last_contact_success`. Even audits may connect to a storage node but count the storage node as offline at the end.

We all agree that the offline tracking function of the satellite is not working right now. That is a different problem for another day.

Reply:
what do you mean by this "We all agree that the offline tracking function of the satellite is not working right now"?  What is not working correctly? Do you just mean how audits don't update `nodes.last_contact_success`?

Reply:
> I feel like we might want the storagenode dashboard to reflect what the satellite thinks about the storagenode, but im not sure.

I fully agree on that. For the moment we have to display some indicator because right now the storage node dashboard will show OFFLINE for everyone. This makes this indicator useless right now. Again this PR is just a quick and dirty solution to show at least an ONLINE status that is correct. If the satellite has a different point of view that we have to fix the satellite and not try to convice the storage node that it should show OFFLINE even if the port forwarding and everything else is working fine. The storage node is handling downloads but the satellite is showing it as offline? Don't blame the storage node for that missmatch.

Reply:
To fix the offline status part, you shouldn't need to add `endpoint.pingStats.WasPinged(time.Now())` in this piecestore file.  pingStats should be updated when the storagenode checks in with the satellite. Are you saying that isn't happening?

Reply:
> Are you saying that isn't happening?

Correct. Please keep in mind that the checkin is once per hour. Any storage node restart before end of hour will reset the dashboard to OFFLINE. 55 minutes later the next checkin will change that but for these 55 minutes the dashboard will show a wrong state.

Reply:
i think the contact.Chore (that checks in with the satellite) should run once when the storagenode first starts up.  That could fix this restart/55 mins problem.  This diff would do that:
```
$ git diff
diff --git a/storagenode/contact/chore.go b/storagenode/contact/chore.go
index b2ba8a35..82db12cc 100644
--- a/storagenode/contact/chore.go
+++ b/storagenode/contact/chore.go
@@ -51,6 +51,9 @@ func (chore *Chore) Run(ctx context.Context) (err error) {
        defer mon.Task()(&ctx)(&err)
        chore.log.Info("Storagenode contact chore starting up")

+       if err := chore.pingSatellites(ctx); err != nil {
+               chore.log.Error("pingSatellites failed", zap.Error(err))
+       }
        return chore.Loop.Run(ctx, func(ctx context.Context) error {
                if err := chore.randomDurationSleep(ctx); err != nil {
                        return err
```

Do you think we should add that?

Reply:
Yes that would fix the issue but to me it sounds like it was implemented different in the first place to prevent storage nodes from DDOS the satellite. @jenlij what do you think?

Reply:
hmm yeah so we added the jitter to prevent a "thundering herd" effect / basically DDOS... We could shorten max wait time

Reply:
Now you lost me. So what can I do to get some kind of fix deployed?

Reply:
i don't like this idea of modifying the pingStats here in piecestore.  What about setting `pingStats.lastPinged` to now when the node starts up instead?